### PR TITLE
feat: ajout d'un nouveau composant ContactForm

### PR DESCRIPTION
**Pull request by franky20000000000**

### CODE REVIEW

File: app/page.tsx
Comment (latioms) [inline app/page.tsx:1]: Prochainement formate le code avant de l'envoyer 

```sh
CTRL + ALT + SHIFT
```

File: components/ui/card.tsx
Comment (latioms) [inline components/ui/card.tsx:1]: Well done

**Review by latioms — APPROVED** [review]

File: app/page.tsx
Comment (franky20000000000) [reply app/page.tsx:1]: ok

**Comment by latioms** [comment]
import React, { useState } from 'react';
import { Card, CardContent, CardHeader, CardTitle } from
'@/components/ui/card';
import { Label } from '@/components/ui/label';
import { Input } from '@/components/ui/input';
import { Button } from '@/components/ui/button';
import { useToast } from "@/components/ui/use-toast"

const ContactForm: React.FC = () => {
  const [fullName, setFullName] = useState('');
  const [phoneNumber, setPhoneNumber] = useState('');
  const [isSubmitting, setIsSubmitting] = useState(false);
  const { toast } = useToast()

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    setIsSubmitting(true);

    try {
      const response = await fetch('/api/register', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ fullname: fullName, phone: phoneNumber }),
      });

      if (response.ok) {
        // Effet de confettis
        import('canvas-confetti').then((confetti) => {
          confetti.default({
            particleCount: 100,
            spread: 70,
            origin: { y: 0.6 }
          });
        });

        toast({
          title: "Succès !",
          description: "Votre contact a été enregistré avec succès.",
        })

        // Réinitialiser le formulaire
        setFullName('');
        setPhoneNumber('');
      } else {
        const errorData = await response.json();
        throw new Error(errorData.error || 'Une erreur est survenue');
      }
    } catch (error) {
      toast({
        title: "Erreur",
        description: error instanceof Error ? error.message : "Une erreur
est survenue lors de l'enregistrement du contact.",
        variant: "destructive",
      })
    } finally {
      setIsSubmitting(false);
    }
  };

  return (
    <Card className="w-[350px]">
      <CardHeader>
        <CardTitle>Formulaire de contact</CardTitle>
      </CardHeader>
      <CardContent>
        <form onSubmit={handleSubmit} className="space-y-8">
          <div className="space-y-2">
            <Label htmlFor="fullName">Nom complet</Label>
            <Input
              id="fullName"
              placeholder="John Doe"
              value={fullName}
              onChange={(e) => setFullName(e.target.value)}
              required
            />
          </div>
          <div className="space-y-2">
            <Label htmlFor="phoneNumber">Numéro de téléphone</Label>
            <Input
              id="phoneNumber"
              placeholder="+33 1 23 45 67 89"
              value={phoneNumber}
              onChange={(e) => setPhoneNumber(e.target.value)}
              required
              pattern="\+?[0-9]{10,14}"
            />
          </div>
          <Button type="submit" disabled={isSubmitting}>
            {isSubmitting ? 'Envoi en cours...' : 'Soumettre'}
          </Button>
        </form>
      </CardContent>
    </Card>
  );
};

export default ContactForm;

Le ven. 6 sept. 2024 à 17:07, Cabrel_Tiomene ***@***.***> a
écrit :

**Comment by latioms** [comment]
npm install --save-dev @types/canvas-confetti

**Pull request merged**